src/UI/Table.elm

```diff
@@ -26,27 +28,45 @@ import UI.Button as Button exposing (Button)
 import UI.Internal.Basics exposing (..)
 import UI.Internal.NList as NList exposing (NList)
 import UI.Internal.Palette as Palette
+import UI.Internal.ToggableList as ToggableList
 import UI.Internal.TypeNumbers as T
 import UI.Palette as Palette exposing (brightnessMiddle, tonePrimary)
-import UI.RenderConfig exposing (RenderConfig)
+import UI.RenderConfig as RenderConfig exposing (RenderConfig)
 import UI.Text as Text exposing (Text)
+import UI.Utils.Element exposing (zeroPadding)
 
 
-type Table msg columns
-    = Table (Properties columns) (Options msg columns)
+type Table msg object columns
+    = Table (Properties columns) (Options msg object columns)
 
 
 type alias Properties columns =
     { headers : HeaderRow columns
     }
 
 
-type alias Options msg columns =
-    { rows : List (Row msg columns)
+type alias Options msg object columns =
+    { rows : Rows msg object columns
     , width : Element.Length
     }
 
 
+type alias ResponsiveOptions msg object columns =
```

Comment:
This case I'm not sure if we should call ResponsiveConfig or Options 🤔 , I think in the pr that I added this idea of having a record for more than `x` fields in a `with*` function I called Config